libs/core/tests/unit_tests/language_models/chat_models/test_base.py

```diff
@@ -654,3 +658,93 @@ def test_normalize_messages_edge_cases() -> None:
         )
     ]
     assert messages == _normalize_messages(messages)
+
+
+def test_streaming_v1() -> None:
+    chunks = [
+        AIMessageChunkV1(
+            [
+                {
+                    "type": "reasoning",
+                    "reasoning": "Let's call a tool.",
```

Comment:
Is there a `is_first_chunk` that mirrors `is_last_chunk`? 


Should we use `chunk_pos: NotRequired[Literal['first', 'last']]`